### PR TITLE
Media component

### PR DESCRIPTION
**Pull request by IbrahimSulai**
Media component is implemented and integrated with our App.

We have used "react-native-video" package to play the media.

**Note:** The video controls support is not available for android. We will take this issue in future.

### CODE REVIEW

File: source/reactnative/src/components/elements/media.js [hunk 1]
```diff
@@ -1,21 +1,96 @@
...
+
+import * as Constants from "../../utils/constants";
+import Video from "react-native-video";
+import ElementWrapper from './element-wrapper'
 
 export class Media extends React.Component {
+    constructor(props) {
+        super(props);
+        this.payload = props.json;
+        this.sources = this.getMediaSources(this.payload.sources)
```
Comment (pragadeeshk) [inline source/reactnative/src/components/elements/media.js]: sources could be undefined. Please Add check in usages

File: source/reactnative/src/components/elements/media.js [hunk 2]
```diff
@@ -1,21 +1,96 @@
...
+    }
+
+    getMediaSources = (sources) => {
+        sources.forEach(source => {
+            this.addUriAttribute(source)
+        })
+        return sources;
+    }
+
+    addUriAttribute = (source) => {
```
Comment (pragadeeshk) [inline source/reactnative/src/components/elements/media.js]: Is there any specific reason to change from url to uri?

File: source/reactnative/src/utils/constants.js [hunk 3]
```diff
@@ -32,6 +32,7 @@ export const PlatformIOS = 'ios';
 export const LightGreyColor = 'lightgrey';
 export const EmphasisColor = '#dcdcdc';
 export const WhiteColor = 'white';
+export const blackColor = 'black';
```
Comment (pragadeeshk) [inline source/reactnative/src/utils/constants.js]: Please follow same naming convention. blackColor -> BlackColor

File: source/reactnative/android/app/src/main/java/com/rnadaptivecards/MainApplication.java [hunk 4]
```diff
@@ -22,7 +23,8 @@ public boolean getUseDeveloperSupport() {
     @Override
     protected List<ReactPackage> getPackages() {
       return Arrays.<ReactPackage>asList(
-          new MainReactPackage()
+          new MainReactPackage(),
+            new ReactVideoPackage()
```
Comment (pragadeeshk) [inline source/reactnative/android/app/src/main/java/com/rnadaptivecards/MainApplication.java]: Use same indentation, either tabs or spaces.

File: source/reactnative/src/components/elements/media.js [hunk 5]
```diff
@@ -1,21 +1,103 @@
...
+        return source;
+    }
+
+    videoError = () => {
+        if (this.state.currentSourceIndex < (this.sources.length - 1)) {
+            this.setState({
+                currentSourceIndex: this.state.currentSourceIndex + 1
+            })
+        }
+    }
```
Comment (pragadeeshk) [inline source/reactnative/src/components/elements/media.js]: Lets propagate the error back to the container to display failure notice.
@BalajiR please comment.

Comment (BalajiR) [reply source/reactnative/src/components/elements/media.js]: Yes @pragadeeshk , already there are error callbacks at root component. Same can be utilized for this case too.
@IbrahimSulai , you can access the error prop from `context` and pass the error details.

File: source/reactnative/src/components/elements/media.js [hunk 6]
```diff
@@ -1,21 +1,114 @@
...
+                                        fullscreen={true}
+                                        controls={true}
+                                        id={this.payload.id ? this.payload.id : "video"}
+                                        paused={false}
+                                        onError={() => { this.videoError(onParseError) }}
+                                        onLoad={this.videoLoadSuccess}
+                                        style={styles.nativeVideoControls}
+                                    />
+                                }
+                                {(!this.state.onLoad && this.payload.poster) && <Image source={{ uri: this.payload.poster }} style={styles.fullScreen}></Image>}
```
[the remark this answers is not in the document]
Comment (IbrahimSulai) [reply source/reactnative/src/components/elements/media.js]: Resolved.

File: source/reactnative/src/components/elements/media.js [hunk 7]
```diff
@@ -1,21 +1,114 @@
...
-
-});
\ No newline at end of file
+    container: {
+        flex: 1,
+        justifyContent: Constants.CenterString,
+        alignItems: Constants.CenterString,
+        backgroundColor: Constants.blackColor,
+    },
+    fullScreen: {
```
[the remark this answers is not in the document]
Comment (IbrahimSulai) [reply source/reactnative/src/components/elements/media.js]: fullscreen style is removed.

File: source/reactnative/src/components/elements/media.js [hunk 8]
```diff
@@ -1,21 +1,114 @@
...
-            </View>
-        )
+            <InputContextConsumer>
+                {({ onParseError }) =>
+                    (
+                        <ElementWrapper json={this.payload}>
+                            <View style={styles.container}>
+                                {
+                                    this.sources &&
+                                    <Video
```
[the remark this answers is not in the document]
Comment (IbrahimSulai) [reply source/reactnative/src/components/elements/media.js]: Yes. <video/> component poster is only working with local URL(internally saved images).  Not working for external URL.

**Pull request merged**